src/util/adb.js

```diff
@@ -311,22 +311,23 @@ export default class ADBUtils {
       value: `-profile ${deviceProfileDir}`,
     }];
 
-    for (const browser of packageIdentifiers) {
-      if (apk.startsWith(browser) && apkComponent) {
-        if (apk !== browser && !apkComponent.includes('.')) {
-          apkComponent = `${browser}.${apkComponent}`;
-        }
-        // if 'geckoview_example' has been found, leave the
-        // loop, before it checks 'geckoview'
-        break;
-      }
-    }
-
     if (!apkComponent) {
       apkComponent = '.App';
     } else if (!apkComponent.includes('.')) {
       apkComponent = `.${apkComponent}`;
     }
+
+    // if `apk` is a browser package or the `apk` has a
+    // browser package prefix: prepend the package identifier
+    // before `apkComponent`
+    if (apkComponent.startsWith('.')) {
+      for (const browser of packageIdentifiers) {
+        if (apk.startsWith(`${browser}.`)) {
```

Comment:
Quick note (before I move on to adding tests):

In the previous comment, there was the condition: 
```
if apk is either identical to the package, or apk starts with `${package}.`
````
For the tests to work, I took out the `apk is identical to the package` part of the condition.

The offended test was: `starts a given APK component without a period`
* Incoming apk: `org.mozilla.geckoview_example`
* Incoming apkComponent: `GeckoViewActivity`
* Expected result: `org.mozilla.geckoview_example/.GeckoViewActivity`
* Result received: `org.mozilla.geckoview_example/org.mozilla.geckoview_example.GeckoViewActivity`

Would you like me to expand out the Expected Result of this test to match?

Reply:
For ADB the two results are equivalent.

The behavior of the received result is probably more desirable to support the .debug/. performancetest scenarios.